lib/logstash/codecs/multiline.rb

```diff
@@ -114,10 +114,23 @@ class LogStash::Codecs::Multiline < LogStash::Codecs::Base
   # to events that actually have multiple lines in them.
   config :multiline_tag, :validate => :string, :default => "multiline"
 
+  # The accumulation of events can make logstash exit with an out of memory error
+  # if event boundaries are not correctly defined. This settings make sure to flush
+  # multiline events after reaching a number of lines, it is used in combination
+  # max_bytes.
+  config :max_lines, :validate => :number, :default => 500
+
+  # The accumulation of events can make logstash exit with an out of memory error
+  # if event boundaries are not correctly defined. This settings make sure to flush
+  # multiline events after reaching a number of bytes, it is used in combination
+  # max_lines.
+  config :max_bytes, :validate => :bytes, :default => "1 MiB"
```

Comment:
I think we should see this as a circuit breaker and thus we can put a high enough limit to make suer we don't interfere with valid multiline payloads. I'd bump that to at least 10Mib